lang/go: Added more features for testing in golang layer

bogdanteleaga opened the pull request:
Adds 3 more functions for golang testing.

During testing the PR I noticed that since I use gocheck it *needs* -check.f not -run, but regular go uses -run, so I included both options.

bogdanteleaga commented:
Looks like (other-window -1) will cycle counter-clockwise. What I'm looking for is the test results to open in another buffer while the focus stays in the buffer that I launched them from. I'm open to suggestions.

syl20bnr commented:
I don't code in Go for now ;-)
Quick question: Have you looked at https://github.com/nlamirault/gotest.el ?

bogdanteleaga commented:
Yup. https://github.com/syl20bnr/spacemacs/pull/3234 :)
The idea of using re-search-backward is taken from there, but I simplified it a bit and added support for suites and gocheck

bogdanteleaga commented:
In case it helps taking a decision, since the layer is using gofmt, the regex *should* never fail, because the function declarations are theoretically always formatted properly.

TheBB commented:
Thank you! Cherry-picked in develop. You can safely delete your branch.
